src/test/java/org/opensearch/neuralsearch/processor/RRFSearchIT.java

```diff
@@ -0,0 +1,171 @@
+        String indexName = "rrf-index";
+        createIndex(indexName, indexSettings, indexMappings, null);
+        addRRFDocuments();
+        createDefaultRRFSearchPipeline();
+
+        Map<String, Object> results = searchRRF(modelId);
+        Map<String, Object> hits = (Map<String, Object>) results.get("hits");
+        ArrayList<HashMap<String, Object>> hitsList = (ArrayList<HashMap<String, Object>>) hits.get("hits");
+        assertEquals(3, hitsList.size());
+        assertEquals(0.016393442, (Double) hitsList.getFirst().get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.016129032, (Double) hitsList.get(1).get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.015873017, (Double) hitsList.getLast().get("_score"), DELTA_FOR_SCORE_ASSERTION);
+    }
+
+    @SneakyThrows
+    private void addRRFDocuments() {
+        addRRFDocument(
+            "A West Virginia university women 's basketball team , officials , and a small gathering of fans are in a West Virginia arena .",
+            "4319130149.jpg"
+        );
+        addRRFDocument("A wild animal races across an uncut field with a minimal amount of trees .", "1775029934.jpg");
+        addRRFDocument(
+            "People line the stands which advertise Freemont 's orthopedics , a cowboy rides a light brown bucking bronco .",
+            "2664027527.jpg"
+        );
+        addRRFDocument("A man who is riding a wild horse in the rodeo is very near to falling off .", "4427058951.jpg");
+        addRRFDocument("A rodeo cowboy , wearing a cowboy hat , is being thrown off of a wild white horse .", "2691147709.jpg");
+    }
+
+    @SneakyThrows
+    private void addRRFDocument(String description, String imageText) {
+        addDocument(RRF_INDEX_NAME, String.valueOf(currentDoc++), "text", description, "image_text", imageText);
+    }
+
+    @SneakyThrows
+    private void createDefaultRRFSearchPipeline() {
+        String requestBody = XContentFactory.jsonBuilder()
+            .startObject()
+            .field("description", "Post processor for hybrid search")
+            .startArray("phase_results_processors")
+            .startObject()
+            .startObject("score-ranker-processor")
+            .startObject("combination")
+            .field("technique", "rrf")
+            .startObject("parameters")
+            .field("rank_constant", 60)
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject()
+            .endArray()
+            .endObject()
+            .toString();
+
+        makeRequest(
+            client(),
+            "PUT",
+            String.format(LOCALE, "/_search/pipeline/%s", RRF_SEARCH_PIPELINE),
+            null,
+            toHttpEntity(String.format(LOCALE, requestBody)),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
+        );
+    }
+
+    @SneakyThrows
+    private Map<String, Object> searchRRF(String modelId) {
```

Comment:
why can't we use existing method from base class, https://github.com/opensearch-project/neural-search/blob/main/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java#L494-L501 or one of it variant. Only customization for RRF is a pipeline name that can be passed as parameter ([code ref](https://github.com/opensearch-project/neural-search/blob/main/src/test/java/org/opensearch/neuralsearch/processor/NormalizationProcessorIT.java#L114))